### PR TITLE
Culture folder, lifestyles and focuses

### LINKED ISSUE #34: First pass implementation for character_backgrounds folder contents
Provide first pass implementation for the **character_backgrounds** folder.

First pass implementation means that all (valid) keys used in vanilla files either:
- Have full working implementations OR
- Have working placeholder implementations (eg. simple scalars) with status documented on the tracker OR
- Have false positives in a placeholder or incomplete implementation that properly covers the majority of use cases. OR
- Have false positives or are unimplementable due to bugs or missing features in cwtools core
        
Additionally, before this issue can be closed, please make sure of the following:
- All significant TODO items are documented in the issue tracker
- Documentation strings are either present or documented as missing in the tracker
        
That being said, these are just the criteria for this issue and scope item to be considered finished. In these early days, every new definition is incredibly helpful for end users trying to use cwtools! PRs with partial implementations are more than welcome!

**This is an auto-generated issue.** It may make sense to close this ticket, combine it with others, or split it into smaller chunks.

### LINKED ISSUE #44: First pass implementation for culture folder contents
Provide first pass implementation for the **culture** folder and the below listed sub-folders.
- [ ] cultures
- [ ] culture_names
- [ ] eras
- [ ] innovations

First pass implementation means that all (valid) keys used in vanilla files either:
- Have full working implementations OR
- Have working placeholder implementations (eg. simple scalars) with status documented on the tracker OR
- Have false positives in a placeholder or incomplete implementation that properly covers the majority of use cases. OR
- Have false positives or are unimplementable due to bugs or missing features in cwtools core
        
Additionally, before this issue can be closed, please make sure of the following:
- All significant TODO items are documented in the issue tracker
- Documentation strings are either present or documented as missing in the tracker
        
That being said, these are just the criteria for this issue and scope item to be considered finished. In these early days, every new definition is incredibly helpful for end users trying to use cwtools! PRs with partial implementations are more than welcome!

**This is an auto-generated issue.** It may make sense to close this ticket, combine it with others, or split it into smaller chunks.

### LINKED ISSUE #62: First pass implementation for focuses folder contents
Provide first pass implementation for the **focuses** folder.

First pass implementation means that all (valid) keys used in vanilla files either:
- Have full working implementations OR
- Have working placeholder implementations (eg. simple scalars) with status documented on the tracker OR
- Have false positives in a placeholder or incomplete implementation that properly covers the majority of use cases. OR
- Have false positives or are unimplementable due to bugs or missing features in cwtools core
        
Additionally, before this issue can be closed, please make sure of the following:
- All significant TODO items are documented in the issue tracker
- Documentation strings are either present or documented as missing in the tracker
        
That being said, these are just the criteria for this issue and scope item to be considered finished. In these early days, every new definition is incredibly helpful for end users trying to use cwtools! PRs with partial implementations are more than welcome!

**This is an auto-generated issue.** It may make sense to close this ticket, combine it with others, or split it into smaller chunks.

### LINKED ISSUE #73: First pass implementation for lifestyles folder contents
Provide first pass implementation for the **lifestyles** folder.

First pass implementation means that all (valid) keys used in vanilla files either:
- Have full working implementations OR
- Have working placeholder implementations (eg. simple scalars) with status documented on the tracker OR
- Have false positives in a placeholder or incomplete implementation that properly covers the majority of use cases. OR
- Have false positives or are unimplementable due to bugs or missing features in cwtools core
        
Additionally, before this issue can be closed, please make sure of the following:
- All significant TODO items are documented in the issue tracker
- Documentation strings are either present or documented as missing in the tracker
        
That being said, these are just the criteria for this issue and scope item to be considered finished. In these early days, every new definition is incredibly helpful for end users trying to use cwtools! PRs with partial implementations are more than welcome!

**This is an auto-generated issue.** It may make sense to close this ticket, combine it with others, or split it into smaller chunks.

### LINKED ISSUE #74: First pass implementation for lifestyle_perks folder contents
Provide first pass implementation for the **lifestyle_perks** folder.

First pass implementation means that all (valid) keys used in vanilla files either:
- Have full working implementations OR
- Have working placeholder implementations (eg. simple scalars) with status documented on the tracker OR
- Have false positives in a placeholder or incomplete implementation that properly covers the majority of use cases. OR
- Have false positives or are unimplementable due to bugs or missing features in cwtools core
        
Additionally, before this issue can be closed, please make sure of the following:
- All significant TODO items are documented in the issue tracker
- Documentation strings are either present or documented as missing in the tracker
        
That being said, these are just the criteria for this issue and scope item to be considered finished. In these early days, every new definition is incredibly helpful for end users trying to use cwtools! PRs with partial implementations are more than welcome!

**This is an auto-generated issue.** It may make sense to close this ticket, combine it with others, or split it into smaller chunks.

### LINKED ISSUE #108: First pass implementation for culture folder contents
Provide first pass implementation for the **history** folder and the below listed sub-folders.
- [ ] culture_names
- [ ] cultures
- [ ] eras
- [ ] innovations

First pass implementation means that all (valid) keys used in vanilla files either:
- Have full working implementations OR
- Have working placeholder implementations (eg. simple scalars) with status documented on the tracker OR
- Have false positives in a placeholder or incomplete implementation that properly covers the majority of use cases. OR
- Have false positives or are unimplementable due to bugs or missing features in cwtools core
        
Additionally, before this issue can be closed, please make sure of the following:
- All significant TODO items are documented in the issue tracker
- Documentation strings are either present or documented as missing in the tracker
        
That being said, these are just the criteria for this issue and scope item to be considered finished. In these early days, every new definition is incredibly helpful for end users trying to use cwtools! PRs with partial implementations are more than welcome!

**This is an auto-generated issue.** It may make sense to close this ticket, combine it with others, or split it into smaller chunks.

### PR DESCRIPTION
**Pull request by LaTueur**

### CODE REVIEW

File: config/enums.cwt [hunk 1]
```diff
@@ -52,4 +52,11 @@ enums = {
         very_high_ai_values_divergence #  350 #About 85% (46-98%) of characters will be at or below this value for the avarage character
         #Percentages in parentheses are the observed spans
     }
+    enum[attributes] = {
```
Comment (Zonr0) [inline config/enums.cwt:55]: These are already referenced as `enum[event_option_skills]` around L100 in events.cwt. This enum is never defined anywhere though. I defined event_option_skills in a PR I'm working on right now, but I can modify it to attributes which makes more sense before I post.

Eitherway, I'd reccomend either changing the enum name to event_option_skills, or changing the relevant line in events.cwt. I'll work around whatever you decide.

File: config/common/culture/culture_names.cwt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+types = {
+    type[culture_name] = {
+        path = "game/common/culture/culture_names"
+    }
+}
+# TODO: Filter _male and _female and actually check if the names exist
```
Comment (Zonr0) [inline config/common/culture/culture_names.cwt:6]: Make sure to file this as an issue after the merge.

File: config/common/culture/cultures.cwt [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+        male_names = single_alias_right[first_name_list]
+        male_names = single_alias_right[first_name_weighted_list]
+
+        female_names = single_alias_right[first_name_list]
+        female_names = single_alias_right[first_name_weighted_list]
+
+        ## cardinality = 0..1
+        ### Chance of male children being named after their paternal grandfather. Sum of these three values must not exceed 100.
+		pat_grf_name_chance = int
```
Comment (Zonr0) [inline config/common/culture/cultures.cwt:68]: Are we sure this and the following fields are int and not `int_value_field`?

File: config/common/culture/era.cwt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    type[era] = {
+        path = "game/common/culture/eras"
+    }
+}
+
+## cardinality = 0..inf
+era = {
+    ## cardinality = 1
+    ### Year when the era can start getting base exposure. Must be 0 or greater.
+    year = int
```
Comment (Zonr0) [inline config/common/culture/era.cwt:11]: Verify `int` or `int_value_field`. I could see this being `int` though.

File: config/common/culture/innovations.cwt [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    unlock_decision = <decision>
+    ## cardinality = 0..inf
+    ### Key of a casus belli that can be unlocked. There can be more than one. This is only to show on the tooltip, it has to be maulually blocked on the object itself.
+    unlock_casus_belli = <casus_belli>
+    ## cardinality = 0..inf
+    ### Key of a regiment that can be unlocked. There can be more than one. Actually does unlock the MaA.
+    unlock_maa = <regiment_type>
+    ## cardinality = 0..inf
+    ### Key of a law that can be unlocked. There can be more than one. This is only to show on the tooltip, it has to be maulually blocked on the object itself.
+    unlock_law = <law>
```
Comment (Zonr0) [inline config/common/culture/innovations.cwt:40]: These _might_ be effects and thus should go in the effects alias, but I don't see them documented in effects.log. Might be worth checking or filing an issue to check to see if these work in other effect contexts.

**Review by Zonr0 — APPROVED** [review]
Great high quality PR that even includes documentation and localisation string info. Make sure to document the todos as issues once merged (feel free to lump related ones together).

Fantastic work, and thanks for the contribution!

File: config/common/culture/innovations.cwt (hunk 5 above)
Comment (LaTueur) [reply config/common/culture/innovations.cwt:40]: They are not in the effects listed by the script_doc command. Also, most of these are just tooltips in reality.

File: config/enums.cwt (hunk 1 above)
Comment (LaTueur) [reply config/enums.cwt:55]: Done in "More documentation, int_value_field and attributes as an enum in events".

File: config/common/culture/cultures.cwt (hunk 3 above)
Comment (LaTueur) [reply config/common/culture/cultures.cwt:68]: Tried it with a value field, throws errors.

File: config/common/culture/era.cwt (hunk 4 above)
Comment (LaTueur) [reply config/common/culture/era.cwt:11]: Tried it with a value field, throws errors.

**Pull request merged**